src/baseclasses/chemical_energy/uvvismeasurementconcentration.py

```diff
@@ -132,14 +131,31 @@ def normalize(self, archive, logger):
             self.figures = [PlotlyFigure(label='figure 1', figure=fig.to_plotly_json())]
 
         if self.chemical_composition_or_formulas and self.peak_value and not self.calibration_measurement:
-            concentration, self.reference = getConcentrationData(archive, logger,
-                                                                 self.chemical_composition_or_formulas,
-                                                                 self.peak_value)
             if self.reference:
-                self.concentration = concentration
-
-
-def getConcentrationData(data_archive, logger, material, peak_value):
+                if isinstance(self.reference, MProxy):
+                    self.reference.m_resolved()
+                    self.concentration = calculate_concentration(self.reference['slope'],
```

Comment:
shouldnt this be one intendion less, so if it is `MProxy` resolve?
I also think you can do it like this `self.reference.slope` this should resolve in place. reference resolving in nomad is weird.

Reply:
I get an `self.concentration = calculate_concentration(self.reference['slope'],
    TypeError: 'MProxy' object is not subscriptable` when I don't `m_resolve` the reference.

Reply:
ah i think you should be able to do `self.reference.slope` that might resolve it in place

Reply:
but it is fine like this too